opensearch/src/auth.rs

```diff
@@ -45,6 +45,13 @@ pub enum Credentials {
     Certificate(ClientCertificate),
     /// An id and api_key to use for API key authentication
     ApiKey(String, String),
+    /// AWS credentials used for AWS SigV4 request signing.
+    /// 
+    /// # Optional
+    /// 
+    /// This requires the `aws-auth` feature to be enabled.
+    #[cfg(feature = "aws-auth")]
+    Aws(aws_types::credentials::SharedCredentialsProvider, aws_types::region::Region)
```

Comment:
should this be `AwsSigV4` in case there is another Aws specific auth method?

Reply:
Happy to rename this. Should I rename the feature as well? Or keep a "generic" `aws-auth` feature and just make the `Credentials` variant more specific?

Reply:
Hmm not sure what is better. If I was using this I would expect the `aws-auth` feature to enable all aws specific auth methods so maybe just making the Credentials variant more specific is fine?